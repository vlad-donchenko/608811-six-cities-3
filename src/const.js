const ZOOM = 12;

const MAX_NEIGHBOURHOOD_OFFERS = 3;

const monthMap = {
  "0": `January`,
  "1": `February`,
  "2": `March`,
  "3": `April`,
  "4": `May`,
  "5": `June`,
  "6": `July`,
  "7": `August`,
  "8": `September`,
  "9": `October`,
  "10": `November`,
  "11": `December`,
};

const OfferPrefix = {
  MAIN: `cities__place-card`,
  DETAILS: `near-places__card`,
};

export {ZOOM, MAX_NEIGHBOURHOOD_OFFERS, OfferPrefix, monthMap};
