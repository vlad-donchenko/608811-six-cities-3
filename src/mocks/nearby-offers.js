const nearbyOffers = [
  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Baby seat`],
    host: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 50,
      isPro: true,
      name: `Angelina`
    },
    id: 50,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: true,
    location: {
      latitude: 52.398298,
      longitude: 4.848328,
      zoom: 8
    },
    maxAdults: 3,
    previewImage: `apartment-01.jpg`,
    price: 424,
    rating: 4,
    title: `Beautiful & luxurious apartment at great location - Nearby`,
    type: `Private room`
  },
  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Baby seat`],
    host: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 51,
      isPro: true,
      name: `Angelina`
    },
    id: 51,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.397288,
      longitude: 4.852012,
      zoom: 8
    },
    maxAdults: 3,
    previewImage: `apartment-01.jpg`,
    price: 124,
    rating: 2,
    title: `Beautiful & luxurious apartment at great location - Nearby`,
    type: `Private room`
  },
  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Baby seat`],
    host: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 52,
      isPro: true,
      name: `Angelina`
    },
    id: 52,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: true,
    isPremium: true,
    location: {
      latitude: 52.395049,
      longitude: 4.854962,
      zoom: 8
    },
    maxAdults: 3,
    previewImage: `apartment-02.jpg`,
    price: 924,
    rating: 2,
    title: `Beautiful & luxurious apartment at great location - Nearby`,
    type: `Private room`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 48.866411,
        longitude: 2.331896,
        zoom: 15
      },
      name: `Paris`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Wi-Fi`, `Kitchen`, `Towels`, `Baby seat`],
    host: {
      avatarUrl: `avatar-max.jpg`,
      id: 53,
      isPro: true,
      name: `Max`
    },
    id: 53,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 48.869508,
      longitude: 2.327385,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `apartment-02.jpg`,
    price: 1100,
    rating: 1,
    title: `Sofitel Paris Le Faubourg - Nearby`,
    type: `Apartment`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 48.866411,
        longitude: 2.331896,
        zoom: 15
      },
      name: `Paris`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Wi-Fi`, `Kitchen`, `Towels`, `Baby seat`],
    host: {
      avatarUrl: `avatar-max.jpg`,
      id: 54,
      isPro: true,
      name: `Max`
    },
    id: 54,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: true,
    isPremium: true,
    location: {
      latitude: 48.870955,
      longitude: 2.330679,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `apartment-02.jpg`,
    price: 100,
    rating: 5,
    title: `Sofitel Paris Le Faubourg - Nearby`,
    type: `Apartment`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 48.866411,
        longitude: 2.331896,
        zoom: 15
      },
      name: `Paris`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Wi-Fi`, `Kitchen`, `Towels`, `Baby seat`],
    host: {
      avatarUrl: `avatar-max.jpg`,
      id: 55,
      isPro: true,
      name: `Max`
    },
    id: 55,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: true,
    location: {
      latitude: 48.869918,
      longitude: 2.327986,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `apartment-03.jpg`,
    price: 190,
    rating: 3,
    title: `Sofitel Paris Le Faubourg - Nearby`,
    type: `Apartment`
  }
];

export default nearbyOffers;
