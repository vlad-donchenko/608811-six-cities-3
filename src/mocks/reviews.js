const reviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 1,
      isPro: false,
      name: `Max`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-06-08T14:13:56.569Z`,
    id: 2,
    rating: 5,
    user: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 2,
      isPro: true,
      name: `Angelina`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-09-08T14:13:56.569Z`,
    id: 3,
    rating: 1,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: false,
      name: `Max`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-10-08T14:13:56.569Z`,
    id: 4,
    rating: 5,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 4,
      isPro: false,
      name: `Angelina`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-10-08T14:13:56.569Z`,
    id: 5,
    rating: 2,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: true,
      name: `Max`
    }
  }
];

export {reviews};
