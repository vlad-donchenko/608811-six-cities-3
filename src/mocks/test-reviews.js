const reviews = [
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-08-08T14:13:56.569Z`,
    id: 1,
    rating: 5,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 1,
      isPro: true,
      name: `Max`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-06-03T14:13:56.569Z`,
    id: 2,
    rating: 4,
    user: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 2,
      isPro: false,
      name: `Angelina`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-06-01T14:13:56.569Z`,
    id: 3,
    rating: 2,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: false,
      name: `Max`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-11-08T14:13:56.569Z`,
    id: 4,
    rating: 2,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 4,
      isPro: false,
      name: `Angelina`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-01-08T14:13:56.569Z`,
    id: 5,
    rating: 4,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: false,
      name: `Max`
    }
  }
];

export {reviews};
