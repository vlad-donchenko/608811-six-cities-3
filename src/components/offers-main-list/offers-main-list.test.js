import React from "react";
import renderer from "react-test-renderer";
import OfferMain from "./offers-main-list.jsx";
import {offers} from "../../mocks/test-offer";
import {OfferPrefix} from "../../const";


it(`Should render <OfferMain/>`, () => {
  const tree = renderer.create(<OfferMain additionalClass={OfferPrefix.CITIES_PLACE_CARD} onOfferMouseLeave={() => {}} onOfferMouseEnter={() => {}} renderOffer={() => {}} offers={offers} onTitleClick={() => {
  }}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
