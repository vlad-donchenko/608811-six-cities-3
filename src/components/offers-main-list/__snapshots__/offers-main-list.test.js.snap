// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <OfferMainList/> 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__place-card__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Beautiful & luxurious studio at great location - 1"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location - 1
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__place-card__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Beautiful & luxurious studio at great location - 2"
          className="place-card__image"
          height="200"
          src="img/apartment-03.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            350
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location - 2
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__place-card__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Beautiful & luxurious studio at great location - 3"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location - 3
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__place-card__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Beautiful & luxurious studio at great location - 4"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            700
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location - 4
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__place-card__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Beautiful & luxurious studio at great location - 5"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            300
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location - 5
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__place-card__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Hotel Düsseldorf City by Tulip Inn"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            160
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Hotel Düsseldorf City by Tulip Inn
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
